x-pack/plugins/aiops/public/components/explain_log_rate_spikes/field_filter_popover.tsx

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState, type ChangeEvent, type FC } from 'react';
+import { css } from '@emotion/react';
+
+import {
+  euiYScrollWithShadows,
+  useEuiTheme,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiPopoverFooter,
+  EuiPopoverTitle,
+  EuiSpacer,
+  EuiSwitch,
+  EuiText,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { FieldFilterApplyButton } from './field_filter_apply_button';
+
+interface FieldFilterPopoverProps {
+  disabled?: boolean;
+  disabledApplyButton?: boolean;
+  uniqueFieldNames: string[];
+  onChange: (skippedFields: string[]) => void;
+}
+
+// This component is mostly inspired by EUI's Data Grid Column Selector
+// https://github.com/elastic/eui/blob/main/src/components/datagrid/controls/column_selector.tsx
+export const FieldFilterPopover: FC<FieldFilterPopoverProps> = ({
+  disabled,
+  disabledApplyButton,
+  uniqueFieldNames,
+  onChange,
+}) => {
+  const euiThemeContext = useEuiTheme();
+  // Inspired by https://github.com/elastic/eui/blob/main/src/components/datagrid/controls/_data_grid_column_selector.scss
+  const fieldSelectPopover = useMemo(
+    () => css`
+      ${euiYScrollWithShadows(euiThemeContext, {})}
+      max-height: 400px;
+    `,
+    [euiThemeContext]
+  );
+
+  const [fieldSearchText, setFieldSearchText] = useState('');
+  const [skippedFields, setSkippedFields] = useState<string[]>([]);
+  const setFieldsFilter = (fieldNames: string[], checked: boolean) => {
+    let updatedSkippedFields = [...skippedFields];
+    if (!checked) {
+      updatedSkippedFields.push(...fieldNames);
+    } else {
+      updatedSkippedFields = skippedFields.filter((d) => !fieldNames.includes(d));
+    }
+    setSkippedFields(updatedSkippedFields);
+  };
+
+  const [isFieldSelectionPopoverOpen, setIsFieldSelectionPopoverOpen] = useState(false);
+  const onFieldSelectionButtonClick = () => setIsFieldSelectionPopoverOpen((isOpen) => !isOpen);
+  const closePopover = () => setIsFieldSelectionPopoverOpen(false);
+
+  const filteredUniqueFieldNames = useMemo(() => {
+    return uniqueFieldNames.filter(
+      (d) => d.toLowerCase().indexOf(fieldSearchText.toLowerCase()) !== -1
+    );
+  }, [fieldSearchText, uniqueFieldNames]);
+
+  // If the supplied list of unique field names changes, do a sanity check to only
+  // keep field names in the list of skipped fields that still are in the list of unique fields.
+  useEffect(() => {
```

Comment:
If you deselect some fields, and then rerun the analysis, for example when changing the deviation / baseline ranges, it looks like the search is using all fields but the popover implies it is reusing the last applied selection. 

<img width="1402" alt="image" src="https://user-images.githubusercontent.com/7405507/230329021-23612263-b7af-4c9b-83d6-d94ebbd5b0a6.png">

For simplicity I think we should do a reset and have all fields selected.

Reply:
Fixed in [45e900b](https://github.com/elastic/kibana/pull/153864/commits/45e900bface874d6e0262e6060c56a72e9559c64).